core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java

```diff
@@ -107,4 +111,50 @@ TransformedClassesBuildItem handleClassTransformation(List<BytecodeTransformerBu
         return new TransformedClassesBuildItem(transformedClassesByJar);
     }
 
+    @BuildStep
+    TransformedArchiveBuildItem archives(ApplicationArchivesBuildItem archives,
+            List<BytecodeTransformerBuildItem> bytecodeTransformerBuildItems) {
+        if (bytecodeTransformerBuildItems.isEmpty()) {
+            return new TransformedArchiveBuildItem(Collections.emptyList(), Collections.emptyMap(), Collections.emptySet());
+        }
+        Set<String> transformedClasses = new HashSet<>();
+        for (BytecodeTransformerBuildItem i : bytecodeTransformerBuildItems) {
+            transformedClasses.add(i.getClassToTransform());
+        }
+        List<ApplicationArchive> transformedArchive = new ArrayList<>();
+        Map<String, Path> applicationClasses = new HashMap<>();
+        Set<String> applicationJavaClassNames = new HashSet<>();
+        try {
+            for (ApplicationArchive aa : archives.getApplicationArchives()) {
+                Path root = aa.getArchiveRoot();
+                Map<String, Path> classes = new HashMap<>();
+                Set<String> javaClassNames = new HashSet<>();
+                AtomicBoolean transform = new AtomicBoolean();
+                Files.walk(root).forEach(new Consumer<Path>() {
```

Comment:
`Files.walk()` doesn't need to be in a try with resources?

I have a vague recollection that it needs to be.